src/libs/ReportUtils.ts

```diff
@@ -3492,6 +3496,72 @@ function buildOptimisticRenamedRoomReportAction(newName: string, oldName: string
     };
 }
 
+/**
+ * Returns the necessary reportAction onyx data to indicate that the transaction has been put on hold optimistically
+ * @param [created] - Action created time
+ */
+function buildOptimisticHoldReportAction(comment: string, created = DateUtils.getDBTime()): OptimisticHoldReportAction {
+    return {
+        reportActionID: NumberUtils.rand64(),
+        actionName: CONST.REPORT.ACTIONS.TYPE.HOLD,
+        pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+        actorAccountID: currentUserAccountID,
+        message: [
+            {
+                type: CONST.REPORT.MESSAGE.TYPE.TEXT,
+                style: 'normal',
+                text: `held this money request with the comment: ${comment}`,
```

Comment:
And we're also missing translate here and at L3548 below that caused https://github.com/Expensify/App/issues/36836.